examples/stencil/src/simd_par.rs

```diff
@@ -0,0 +1,88 @@
+//! SIMD+Rayon implementation.
+
+use super::simd::step_x8;
+use rayon::prelude::*;
+
+pub fn x8_par(
+    t0: i32,
+    t1: i32,
+    x0: i32,
+    x1: i32,
+    y0: i32,
+    y1: i32,
+    z0: i32,
+    z1: i32,
+    n_x: i32,
+    n_y: i32,
+    n_z: i32,
+    coef: &[f32; 4],
+    vsq: &[f32],
+    a_even: &mut [f32],
+    a_odd: &mut [f32],
+) {
+    let a_even = a_even.as_mut_ptr() as isize;
+    let a_odd = a_odd.as_mut_ptr() as isize;
+    for t in t0..t1 {
+        (z0..z1).into_par_iter().for_each(|z| {
+            let (a_even, a_odd) = unsafe {
+                let n = (n_x * n_y * n_z) as usize;
+                let a_even: &mut [f32] =
+                    ::std::slice::from_raw_parts_mut(a_even as *mut f32, n);
+                let a_odd: &mut [f32] =
+                    ::std::slice::from_raw_parts_mut(a_odd as *mut f32, n);
+                (a_even, a_odd)
```

Comment:
@cuviper this is undefined behavior because it creates multiple `&mut [f32]` that reference the exact same slice. I couldn't find a way to chunk this properly.

Reply:
So i cleaned this up a bit in the last commit, but it still has undefined behavior, and the main reason I can clean it up is that if I split the matrix  into slices that each call write to, then I'll need to index those slices with different indices than the other matrix that I only read from.

I'll try to do that later, but I don't think i will be able to reuse the serial code for this.